R/createSets.R

```diff
@@ -0,0 +1,267 @@
+         "There has to be at least one historic period.")
+  }
+
+  invisible(m$addSet(
+    "tall",
+    records = min(ttotNum):max(ttotNum),
+    description = "all time steps"
+  ))
+
+  ttot <- m$addSet(
+    "ttot",
+    records = ttotNum,
+    description = "all modelling time steps"
+  )
+  invisible(m$addAlias("ttot2", ttot))
+
+  invisible(m$addSet(
+    "tinit",
+    records = min(ttotNum),
+    description = "initial modelling time step"
+  ))
+  t <- m$addSet(
+    "t",
+    records = ttot$getUELs()[which(ttot$getUELs() >= startyear)],
+    description = "modelled time steps"
+  )
+
+  invisible(m$addSet(
+    "thist",
+    records = setdiff(ttot$getUELs(), t$getUELs()),
+    description = "historic time steps"
+  ))
+
+
+  # Vintages -------------------------------------------------------------------
+
+  vintages <- getBrickMapping("vintage.csv")
+
+  vin <- m$addSet(
+    "vin",
+    records = unique(getElement(vintages, "vin")),
+    description = "construction vintage cohort"
+  )
+
+  vinExists <- expandSets(ttot, vin) %>%
+    left_join(vintages, by = "vin") %>%
+    filter(.data[["ttot"]] > .data[["from"]] - 1) %>%
+    select("ttot", "vin")
+  vinExists <- m$addSet(
+    "vinExists",
+    c(ttot, vin),
+    records = vinExists,
+    description = "Can this vintage cohort exist i.e. ttot cannot be before cohort starts"
+  )
+
+
+
+  # Building state alternatives ------------------------------------------------
+
+  ## building shell ====
+
+  bs <- getBrickMapping("buildingShell.csv") %>%
+    getElement("bs") %>%
+    unique()
+  if (config[["ignoreShell"]]) bs <- head(bs, 1)
+  bs <-  m$addSet(
+    "bs",
+    records = bs,
+    description = "building shell"
+  )
+
+  bsr <- m$addSet(
+    "bsr",
+    records = c(bs$getUELs(), 0),
+    description = "renovated building shell"
+  )
+
+
+  ## heating system ====
+
+  hsMap <- getBrickMapping("heatingSystem.csv", "sectoral")
+
+  hs <- hsMap %>%
+    getElement("hs") %>%
+    unique()
+  hs <-  m$addSet(
+    "hs",
+    records = hs,
+    description = "heating system"
+  )
+
+  hsr <- m$addSet(
+    "hsr",
+    records = c(0, hs$getUELs()),
+    description = "renovated heating system"
+  )
+
+  carrier <- hsMap %>%
+    getElement("carrier") %>%
+    unique()
+  carrier <- m$addSet(
+    "carrier",
+    records = carrier,
+    description = "energy carrier"
+  )
+
+  hsCarrier <- unique(hsMap[, c("hs", "carrier")])
+  hsCarrier <- m$addSet(
+    "hsCarrier",
+    c(hs, carrier),
+    records = hsCarrier,
+    description = "mapping between heating system and energy carrier"
+  )
+
+
+
+  # Independent stock subset ---------------------------------------------------
+
+  reg <- m$addSet(
+    "reg",
+    records = config[["regions"]],
+    description = "region"
+  )
+
+  loc <- getBrickMapping("location.csv") %>%
+    getElement("loc") %>%
+    unique()
+  loc <- m$addSet(
+    "loc",
+    records = loc,
+    description = "location of building (rural, urban)"
+  )
```

Comment:
As far as I can see, you don't need the set ```loc``` as an R variable here, same applies to ```typ``` and ```inc``` below. But it might make sense to overwrite the previous data frame?

Reply:
Yeah, it doesn't really matter if I overwrite it or use `invisible` to avoid this. If there was no such object before and I create it with adding a symbol to the gams Container, linter will complain about an object not being used. Therefore, I introduced `invisible`. I think that I don't want to do this if not demanded by linter so I can use the gams symbols in later calculations without changing lines above if I find out that I need them later. But if you think differently, I'm happy to change it.

Reply:
No, I fully agree, your argumentation makes perfect sense to me - keep the lintr happy and still produce easily adaptable code :-)